pressbooks/pressbooks
# Replace and notify invalidated Bisac codes

Related issue: https://github.com/pressbooks/pressbooks-bisac/issues/6

This change add an alert at the top of the edit book form when a invalidated Bisac code is found and replace those by using a new filter added in https://github.com/pressbooks/pressbooks-bisac/pull/9
Invalidated bisac codes are in: https://bisg.org/page/InactivatedCodes and the replacements suggested internally here: https://docs.google.com/spreadsheets/d/1-KL0xnLVM-ELLnHdV-Z4BiykJaKMblY2/edit#gid=2042444657 
Since we update our Bisac subject with the 2020 edition, those invalidated codes could be present in some books, but we are not able to display them in the interface.
With this message we invite those book authors to change the Bisac codes.

### How to test
Requirement: PB Bisac plugin activated.
- Insert a invalidated Bisac code in the database. Those are in 2 tables: `wp_blogmeta` and `wp_<blog_id>_postmeta` tables.
Example: Inserting the `REF028000` Bisac code (invalidated) for blog_id = 2
```
update wp_blogmeta set meta_value='a:13:{s:10:"pb_authors";s:5:"admin";s:10:"pb_editors";s:0:"";s:14:"pb_translators";s:0:"";s:12:"pb_reviewers";s:0:"";s:15:"pb_illustrators";s:0:"";s:15:"pb_contributors";s:0:"";s:8:"pb_title";s:22:"The onboarding process";s:11:"pb_language";s:2:"en";s:14:"pb_cover_image";s:88:"https://pressbooks.test/app/plugins/pressbooks/assets/dist/images/default-book-cover.jpg";s:18:"pb_primary_subject";s:3:"AVR";s:23:"pb_bisac_regional_theme";s:13:"1.1.1.0.0.0.0";s:22:"pb_additional_subjects";s:4:"YPAD";s:16:"pb_bisac_subject";s:20:"REF028000";}' where meta_id = 32 and blog_id = 2
update wp_2_postmeta set meta_value = 'REF028000' where meta_id = <Check Meta ID> and post_id = 16\G
```
- Alternatively you can checkout to a previous version of pressbooks-bisac plugin and update a book with a invalidated Bisac code, and then move to the latest dev version of the plugin (or to this PR: https://github.com/pressbooks/pressbooks-bisac/pull/9).
- Now you can try to edit the book, and you should be able to see the notice at the top of the form.
- New unit tests can be executed running `vendor/bin/phpunit --configuration phpunit.xml --group book`
